### ISSUE TITLE
False positive in `undefined-all-variable` and unable to ignore it

### ISSUE THREAD

**Issue by davfsa** [PyCQA/pylint#3469]
### Steps to reproduce
1. Create a file with this code:
```py
__all__ = ["HikariError"]  # pylint: disable=undefined-all-variable


class HikariError(RuntimeError):
    """Base for an error raised by this API."""

    __slots__ = ()
```
2. Run pylint to check the file
3. Observe

### Current behavior
Errors and cant be ignored

### Expected behavior
Doesnt error and it can be ignored if needed

### pylint --version output
pylint 2.4.4
astroid 2.3.3
Python 3.8.0 (default, Oct 28 2019, 16:14:01) 
[GCC 9.2.1 20191008]

### Related issue
#126

**Comment by PCManticore** [comment]
Thanks for the report. This does not reproduce for me using your snippet and the following versions:
```
pylint 2.4.4
astroid 2.3.3
Python 3.8.0 (default, Nov  6 2019, 18:25:08)
[Clang 8.1.0 (clang-802.0.42)]
```

I also tried with the unreleased 2.5.0 version and doesn't reproduce as well. Is this error reproducing for you using this snippet or was there some additional code?

**Comment by davfsa** [comment]
@PCManticore  After you closed this I have been trying to figure out why. I have only been able to reproduce this is if you name the file `errors.py`.
